# 算术异常 ArithmeticException

发现一个有趣的现象:

```
public static  double devide(double a,double b){

    try {
        LOGGER.info("a"+"/ b"+"="+(a/b));
        return  a/b;
    }
    catch (ArithmeticException e){
        LOGGER.error("错误:"+e.getMessage());
        return  0;
    }
}
```

在此例题中，即使b为0，也不会进入catch中，因为double算术中，如果b为0，这个除法会认为得到一个无限大的数。会得到一个 a/ b=Infinity，这是个无穷的值。

如果在int类型中，会出现错误，进入catch中。错误:/ by zero，不能为0.可以用如下的LOOGER.warn来判断一下，

```
public static  int devide(int a,int b){

    try {
        if (b==0) {
            LOGGER.warn("被除数不能为0");
            return 0;
        }
        else {
            LOGGER.info("a" + "/ b" + "=" + (a / b));
            return a / b;
        }
    }
    catch (ArithmeticException e){
        LOGGER.error("错误:"+e.getMessage());
        return  0;
    }
}
```